libs/common-services/src/main/java/uk/gov/di/ipv/core/library/domain/JourneyRequest.java

```diff
@@ -14,5 +14,6 @@ public class JourneyRequest {
     private String ipAddress;
     private String clientOAuthSessionId;
     private String journey;
+    private String context;
```

Comment:
I think this adds `context` to _all_ journey requests, rather than just ones to the `BuildCriOauthRequestHandler` - is that intentional?

Not necessarily a problem, and context could be useful for others in future, but a bit different to what we've done elsewhere. Compare the `ProcessRequest`, which is specific request class for the check GPG45 score lambda.

Reply:
I'm on the fence about adding a new class for the request with context since it isn't particularly specialised, but I definitely can if you think it's worth while

Reply:
I'd be inclined to not add a new class. The `ProcessRequest` class isn't really implemented how I thought it should be - it's not generic. I'd intended that you'd be able to use that class for any "process" type state and freely define data to pass through, rather than having to update the class every time you wanted to use it to run a different lambda.

The new `context` attribute is only being used on the cri type states, which doesn't use the `ProcessRequest` state. I don't think I'd want to see a proliferation of classes for each bit of data we want to pass through from the journey map. And as you say, "context" is pretty generic and could be used for other things.

Reply:
Bear in mind that we're also about to add `scope` which is a bit more CRI-specific, so instead we're going to end up with a proliferation of fields that are only applicable to certain steps...

On balance, I think I'd slightly prefer a separate class, so you know exactly what each lambda expects, but I'm happy for you to leave if you'd like to move on.

cc @rpayal - this will affect you as well!